### ISSUE TITLE
id_ function for vectors

### ISSUE THREAD

**Issue by tueda** [vermaseren/form#155]
It would be nice to extend the functionality of `id_` also for vectors.
```
V p1,p2,p3,p4;
L F = p1.p2 * id_(p1,p3);
P;
.end
```
currently gives
```
FORM 4.1 (Nov 29 2016, v4.1-20131025-272-gb993ca7) 64-bits  Run: Mon Dec  5 13:59:58 2016

   F =
      id_(p1,p3)*p1.p2;
```
This is an easy way to dynamically replace dot products in numerators (including squared ones) while untouching denominators. (`replace_` affects on both numerators and denominators).

**Comment by tueda** [comment]
Might be nice also for functions as
```
CF f(s);
L F = f(1,2,3);
multiply id_(f,dummy_);
* id f(?a) = 1;                 * compile-time error
* multiply replace_(f,dummy_);  * dangerous, not only the ground level
P;
.end
```
though I don't know the exact meaning of `dummy_` yet (#143).

**Comment by vermaseren** [comment]
And how would you do that. For functions
   id f = dummy_;
is different from
   id f(?a) = dummy_(?a);
Because id_ is a function
   Multiply id_(f(?a),dummy_(?a));
Cannot work either, because ?a is not defined in a LHS.
In general the id_ cannot work with wiuldcards in the same way as the
id statement can.
With normal wildcard variables one could envision
   Multiply id_(x,y,f(x,y),dummy_(x,y));
to mean that x and y should be seen as wildcards in the LHS that will be
generated, but this still excludes the use of ?a type wildcards.
Anyway, in your example f is a symmetric function and f(?a) cannot
work either in the current pattern matcher.

Suggestions?

Jos

**Comment by tueda** [comment]
You are right for `id_` for functions. Actually what one needs (and may expect for `id`_) is `replace_` with specifying the level where the substitution is performed.
